lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -0,0 +1,43 @@
+module CC
+  module Engine
+    module Analyzers
+      class AnalyzerBase
+        def initialize(engine_config:, directory:)
+          @engine_config = engine_config
+          @directory = directory
+        end
+
+        def run
+          files.map do |file|
+            process_file(file)
+          end
+        end
+
+        def mass_threshold
+          engine_config.mass_threshold_for(self.class::LANGUAGE) || self.class::DEFAULT_MASS_THRESHOLD
+        end
+
+        def base_points
+          self.class::BASE_POINTS
+        end
+
+        private
+
+        attr_reader :engine_config, :directory
+
+        def process_file(path)
+          raise "Subclass must implement `process_file`"
```

Comment:
Ah interesting. We use it elsewhere as a convention for this abstract class purpose. I think `NoMethodError` would be incorrect. Maybe just best to leave it as-is